diff -burN ecore-1.7.9/configure.ac ecore-1.7.9-PS3/configure.ac
--- ecore-1.7.9/configure.ac	2013-10-31 18:38:05.000000000 -0400
+++ ecore-1.7.9-PS3/configure.ac	2014-07-02 13:54:25.600586320 -0400
@@ -167,6 +167,7 @@
 want_ecore_evas_gl_sdl="no"
 want_ecore_evas_gl_cocoa="no"
 want_ecore_evas_psl1ght="no"
+want_ecore_evas_gl_psl1ght="no"
 want_ecore_evas_directfb="no"
 want_ecore_evas_fb="no"
 want_ecore_evas_software_16_wince="no"
@@ -473,6 +474,7 @@
       requirements_ecore_sdl="escape ${requirements_ecore_sdl}"
       want_ecore_psl1ght="yes"
       want_ecore_evas_psl1ght="yes"
+      want_ecore_evas_gl_psl1ght="yes"
       ;;
 esac
 
@@ -1934,6 +1936,14 @@
    [${have_ecore_psl1ght}],
    [requirements_ecore_evas="ecore-psl1ght >= 1.7.9 ${requirements_ecore_evas}"])
 
+# ecore_evas_gl_psl1ght
+
+ECORE_EVAS_CHECK_MODULE([gl-psl1ght],
+   [${want_ecore_evas_gl_psl1ght}],
+   [GL_PSL1GHT],
+   [${have_ecore_psl1ght}],
+   [requirements_ecore_evas="ecore-psl1ght >= 1.7.9 ${requirements_ecore_evas}"])
+
 # ecore_imf_ibus
 PKG_CHECK_MODULES([IBUS], [ibus-1.0 >= 1.4], [have_ibus="yes"], [have_ibus="no"])
 
@@ -2231,6 +2241,7 @@
   echo "    OpenGL Cocoa...............: $have_ecore_evas_opengl_cocoa"
   echo "    DirectFB...................: $have_ecore_evas_directfb"
   echo "    PSL1GHT....................: $have_ecore_evas_psl1ght"
+  echo "    RSXGL PSL1GHT..............: $have_ecore_evas_gl_psl1ght"
   echo "    Software Framebuffer.......: $have_ecore_evas_fb"
   echo "    Software 8bit grayscale XCB: $have_ecore_evas_software_8_x11"
   echo "    Software 16bit X11.........: $have_ecore_evas_software_16_x11"
diff -burN ecore-1.7.9/src/lib/ecore_psl1ght/moveutil.h ecore-1.7.9-PS3/src/lib/ecore_psl1ght/moveutil.h
--- ecore-1.7.9/src/lib/ecore_psl1ght/moveutil.h	1969-12-31 19:00:00.000000000 -0500
+++ ecore-1.7.9-PS3/src/lib/ecore_psl1ght/moveutil.h	2014-07-02 13:55:33.445588731 -0400
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) Antonio José Ramos Márquez (bigboss)
+ * Copyright (C) Youness Alaoui (KaKaRoTo)
+ */
+
+#ifndef __MOVEUTIL_H__
+#define __MOVEUTIL_H__
+
+#include <ppu-types.h>
+#include <spurs/spurs.h>
+#include <io/camera.h>
+#include <io/move.h>
+
+typedef struct
+{
+   Spurs         *spurs;
+   cameraInfoEx   camInfo;
+   cameraReadInfo camRead;
+   gemState       state;
+} moveContext;
+
+static inline float
+vec_array(vec_float4 vec, unsigned int idx)
+{
+   union {
+      vec_float4 vec;
+      float      array[4];
+   } v;
+
+   v.vec = vec;
+
+   if (idx > 3)
+     return -1;
+   return v.array[idx];
+}
+
+moveContext *initMove();
+void         endMove(moveContext *context);
+void         readGemState(moveContext *context);
+int          processMove(moveContext *context);
+void         moveGet3DPosition(moveContext *context, float *x, float *y, float *z);
+
+#endif /* __MOVEUTIL_H__ */
diff -burN ecore-1.7.9/src/lib/ecore_psl1ght/spursutil.h ecore-1.7.9-PS3/src/lib/ecore_psl1ght/spursutil.h
--- ecore-1.7.9/src/lib/ecore_psl1ght/spursutil.h	1969-12-31 19:00:00.000000000 -0500
+++ ecore-1.7.9-PS3/src/lib/ecore_psl1ght/spursutil.h	2014-07-02 13:55:33.446588732 -0400
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) Antonio José Ramos Márquez (bigboss)
+ * Copyright (C) Youness Alaoui (KaKaRoTo)
+ */
+
+#ifndef __SPURSUTIL_H__
+#define __SPURSUTIL_H__
+
+#include <ppu-types.h>
+#include <spurs/spurs.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Initialize spurs with a given name prefix (*/
+Spurs *initSpurs(const char *prefix_name);
+void   endSpurs(Spurs *spurs);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* __SPURSUTIL_H__ */
