diff -burN ecore-1.7.9/configure.ac ecore-1.7.9-PS3/configure.ac
--- ecore-1.7.9/configure.ac	2013-10-31 18:38:05.000000000 -0400
+++ ecore-1.7.9-PS3/configure.ac	2014-07-03 23:31:59.213888968 -0400
@@ -167,6 +167,7 @@
 want_ecore_evas_gl_sdl="no"
 want_ecore_evas_gl_cocoa="no"
 want_ecore_evas_psl1ght="no"
+want_ecore_evas_gl_psl1ght="no"
 want_ecore_evas_directfb="no"
 want_ecore_evas_fb="no"
 want_ecore_evas_software_16_wince="no"
@@ -473,6 +474,7 @@
       requirements_ecore_sdl="escape ${requirements_ecore_sdl}"
       want_ecore_psl1ght="yes"
       want_ecore_evas_psl1ght="yes"
+      want_ecore_evas_gl_psl1ght="yes"
       ;;
 esac
 
@@ -1908,6 +1910,22 @@
    [Ecore Evas Single Process Windowing System],
    [yes], [])
 
+# ecore_evas_psl1ght
+
+ECORE_EVAS_CHECK_MODULE([psl1ght],
+   [${want_ecore_evas_psl1ght}],
+   [PSL1GHT],
+   [${have_ecore_psl1ght}],
+   [requirements_ecore_evas="ecore-psl1ght >= 1.0.0 ${requirements_ecore_evas}"])
+
+# ecore_evas_gl_psl1ght
+
+ECORE_EVAS_CHECK_MODULE([gl-psl1ght],
+   [${want_ecore_evas_gl_psl1ght}],
+   [GL_PSL1GHT],
+   [${have_ecore_psl1ght}],
+   [requirements_ecore_evas="ecore-psl1ght >= 1.6.99 ${requirements_ecore_evas}"])
+
 # ecore_evas_extn
 
 have_extn="yes"
@@ -1934,6 +1952,14 @@
    [${have_ecore_psl1ght}],
    [requirements_ecore_evas="ecore-psl1ght >= 1.7.9 ${requirements_ecore_evas}"])
 
+# ecore_evas_gl_psl1ght
+
+ECORE_EVAS_CHECK_MODULE([gl-psl1ght],
+   [${want_ecore_evas_gl_psl1ght}],
+   [GL_PSL1GHT],
+   [${have_ecore_psl1ght}],
+   [requirements_ecore_evas="ecore-psl1ght >= 1.7.9 ${requirements_ecore_evas}"])
+
 # ecore_imf_ibus
 PKG_CHECK_MODULES([IBUS], [ibus-1.0 >= 1.4], [have_ibus="yes"], [have_ibus="no"])
 
@@ -2231,6 +2257,7 @@
   echo "    OpenGL Cocoa...............: $have_ecore_evas_opengl_cocoa"
   echo "    DirectFB...................: $have_ecore_evas_directfb"
   echo "    PSL1GHT....................: $have_ecore_evas_psl1ght"
+  echo "    RSXGL PSL1GHT..............: $have_ecore_evas_gl_psl1ght"
   echo "    Software Framebuffer.......: $have_ecore_evas_fb"
   echo "    Software 8bit grayscale XCB: $have_ecore_evas_software_8_x11"
   echo "    Software 16bit X11.........: $have_ecore_evas_software_16_x11"
diff -burN ecore-1.7.9/src/lib/ecore/ecore.c ecore-1.7.9-PS3/src/lib/ecore/ecore.c
--- ecore-1.7.9/src/lib/ecore/ecore.c	2013-07-29 10:22:11.000000000 -0400
+++ ecore-1.7.9-PS3/src/lib/ecore/ecore.c	2014-07-03 23:14:40.752852057 -0400
@@ -259,8 +259,11 @@
     */
      p = _thread_call;
      _thread_call = NULL;
-     _ecore_pipe_wait(p, 1, 0.1);
-     _ecore_pipe_del(p);
+     if (p)
+       {
+         ecore_pipe_wait(p, 1, 0.1);
+         ecore_pipe_del(p);
+       }
      eina_lock_free(&_thread_safety);
      eina_condition_free(&_thread_cond);
      eina_lock_free(&_thread_mutex);
diff -burN ecore-1.7.9/src/lib/ecore/ecore_pipe.c ecore-1.7.9-PS3/src/lib/ecore/ecore_pipe.c
--- ecore-1.7.9/src/lib/ecore/ecore_pipe.c	2013-07-29 10:22:11.000000000 -0400
+++ ecore-1.7.9-PS3/src/lib/ecore/ecore_pipe.c	2014-07-03 23:14:40.752852057 -0400
@@ -461,6 +461,11 @@
    int total = 0;
 
    EINA_MAIN_LOOP_CHECK_RETURN_VAL(-1);
+   if (!ECORE_MAGIC_CHECK(p, ECORE_MAGIC_PIPE))
+     {
+        ECORE_MAGIC_FAIL(p, ECORE_MAGIC_PIPE, "ecore_pipe_wait");
+        return -1;
+     }
    if (p->fd_read == PIPE_FD_INVALID)
      return -1;
 
diff -burN ecore-1.7.9/src/lib/ecore_evas/ecore_evas.c ecore-1.7.9-PS3/src/lib/ecore_evas/ecore_evas.c
--- ecore-1.7.9/src/lib/ecore_evas/ecore_evas.c	2013-07-29 10:22:11.000000000 -0400
+++ ecore-1.7.9-PS3/src/lib/ecore_evas/ecore_evas.c	2014-07-03 23:14:40.753852057 -0400
@@ -201,6 +201,12 @@
 #else
         return EINA_FALSE;
 #endif
+     case ECORE_EVAS_ENGINE_GL_PSL1GHT:
+#ifdef BUILD_ECORE_EVAS_GL_PSL1GHT
+        return EINA_TRUE;
+#else
+        return EINA_FALSE;
+#endif
      case ECORE_EVAS_ENGINE_WAYLAND_SHM:
 #ifdef BUILD_ECORE_EVAS_WAYLAND_SHM
         return EINA_TRUE;
@@ -601,6 +607,22 @@
 }
 #endif
 
+#ifdef BUILD_ECORE_EVAS_GL_PSL1GHT
+static Ecore_Evas *
+_ecore_evas_constructor_gl_psl1ght(int x __UNUSED__, int y __UNUSED__, int w, int h, const char *extra_options)
+{
+   Ecore_Evas *ee;
+   char *name = NULL;
+
+   _ecore_evas_parse_extra_options_str(extra_options, "name=", &name);
+   ee = ecore_evas_gl_psl1ght_new(name, w, h);
+   free(name);
+
+   if (ee) ecore_evas_move(ee, x, y);
+   return ee;
+}
+#endif
+
 #ifdef BUILD_ECORE_EVAS_WAYLAND_SHM
 static Ecore_Evas *
 _ecore_evas_constructor_wayland_shm(int x, int y, int w, int h, const char *extra_options)
@@ -770,6 +792,9 @@
 #endif
 
   /* PS3 support */
+#ifdef BUILD_ECORE_EVAS_GL_PSL1GHT
+  {"gl_psl1ght", _ecore_evas_constructor_gl_psl1ght},
+#endif
 #ifdef BUILD_ECORE_EVAS_PSL1GHT
   {"psl1ght", _ecore_evas_constructor_psl1ght},
 #endif
diff -burN ecore-1.7.9/src/lib/ecore_evas/Ecore_Evas.h ecore-1.7.9-PS3/src/lib/ecore_evas/Ecore_Evas.h
--- ecore-1.7.9/src/lib/ecore_evas/Ecore_Evas.h	2013-07-29 10:22:11.000000000 -0400
+++ ecore-1.7.9-PS3/src/lib/ecore_evas/Ecore_Evas.h	2014-07-03 23:34:43.309894800 -0400
@@ -100,6 +100,7 @@
 #define HAVE_ECORE_EVAS_WINCE 1
 #define HAVE_ECORE_EVAS_EWS 1
 #define HAVE_ECORE_EVAS_PSL1GHT 1
+#define HAVE_ECORE_EVAS_GL_PSL1GHT 1
 #define HAVE_ECORE_EVAS_WAYLAND_SHM 1
 #define HAVE_ECORE_EVAS_WAYLAND_EGL 1
 
@@ -127,7 +128,8 @@
    ECORE_EVAS_ENGINE_EWS,
    ECORE_EVAS_ENGINE_PSL1GHT,
    ECORE_EVAS_ENGINE_WAYLAND_SHM,
-   ECORE_EVAS_ENGINE_WAYLAND_EGL
+   ECORE_EVAS_ENGINE_WAYLAND_EGL,
+   ECORE_EVAS_ENGINE_GL_PSL1GHT
 } Ecore_Evas_Engine_Type;
 
 typedef enum _Ecore_Evas_Avoid_Damage_Type
@@ -1159,6 +1161,7 @@
 				      int h);
 
 EAPI Ecore_Evas *ecore_evas_psl1ght_new(const char* name, int w, int h);
+EAPI Ecore_Evas *ecore_evas_gl_psl1ght_new(const char* name, int w, int h);
 
 
 /* generic manipulation calls */
diff -burN ecore-1.7.9/src/lib/ecore_evas/ecore_evas_psl1ght.c ecore-1.7.9-PS3/src/lib/ecore_evas/ecore_evas_psl1ght.c
--- ecore-1.7.9/src/lib/ecore_evas/ecore_evas_psl1ght.c	2013-07-29 10:22:11.000000000 -0400
+++ ecore-1.7.9-PS3/src/lib/ecore_evas/ecore_evas_psl1ght.c	2014-07-03 23:14:40.755852057 -0400
@@ -21,6 +21,7 @@
 };
 
 static const char *ecore_evas_psl1ght_default = "EFL PSL1GHT";
+static const char *ecore_evas_gl_psl1ght_default = "EFL RSXGL PSL1GHT";
 static int _ecore_evas_fps_debug = 0;
 static Ecore_Poller *ecore_evas_event;
 
@@ -409,8 +410,8 @@
    NULL  // screen_dpi_get
 };
 
-EAPI Ecore_Evas *
-ecore_evas_psl1ght_new(const char *name, int w, int h)
+static Ecore_Evas *
+_ecore_evas_psl1ght_new(const char *name, const char *engine, int w, int h)
 {
    void *einfo;
    Ecore_Evas *ee;
@@ -425,7 +426,7 @@
 
    ee->engine.func = (Ecore_Evas_Engine_Func *)&_ecore_psl1ght_engine_func;
 
-   ee->driver = "psl1ght";
+   ee->driver = engine;
    if (name) ee->name = strdup(name);
 
    if (w < 1) w = 1;
@@ -449,7 +450,7 @@
    /* init evas here */
    ee->evas = evas_new();
    evas_data_attach_set(ee->evas, ee);
-   evas_output_method_set(ee->evas, evas_render_method_lookup("psl1ght"));
+   evas_output_method_set(ee->evas, evas_render_method_lookup(engine));
 
    evas_output_size_set(ee->evas, w, h);
    evas_output_viewport_set(ee->evas, 0, 0, w, h);
@@ -503,6 +504,18 @@
    return ee;
 }
 
+EAPI Ecore_Evas *
+ecore_evas_psl1ght_new(const char *name, int w, int h)
+{
+  return _ecore_evas_psl1ght_new (name, "psl1ght", w, h);
+}
+
+EAPI Ecore_Evas *
+ecore_evas_gl_psl1ght_new(const char *name, int w, int h)
+{
+  return _ecore_evas_psl1ght_new (name, "gl_psl1ght", w, h);
+}
+
 #else /* BUILD_ECORE_EVAS_PSL1GHT */
 
 EAPI Ecore_Evas *
@@ -510,6 +523,13 @@
 {
    ERR("OUTCH !");
    return NULL;
+}
+
+EAPI Ecore_Evas *
+ecore_evas_gl_psl1ght_new(const char *name __UNUSED__, int w __UNUSED__, int h __UNUSED__)
+{
+   ERR("OUTCH !");
+   return NULL;
 }
 
 #endif /* BUILD_ECORE_EVAS_PSL1GHT */
diff -burN ecore-1.7.9/src/lib/ecore_psl1ght/gemutil.c ecore-1.7.9-PS3/src/lib/ecore_psl1ght/gemutil.c
--- ecore-1.7.9/src/lib/ecore_psl1ght/gemutil.c	1969-12-31 19:00:00.000000000 -0500
+++ ecore-1.7.9-PS3/src/lib/ecore_psl1ght/gemutil.c	2014-07-03 23:25:44.303875642 -0400
@@ -0,0 +1,281 @@
+#include <stdio.h>
+#include <io/move.h>
+#include <sys/memory.h>
+#include <ppu-types.h>
+#include <io/camera.h>
+#include <ppu-types.h>
+#include <spurs/spurs.h>
+#include <sys/spu.h>
+#include <sys/thread.h>
+#include <sys/systime.h>
+
+#define SPURS_DEFAULT_PREFIX_NAME "gem_spurs"
+
+u16 oldGemPad = 0;
+u16 newGemPad = 0;
+u16 newGemAnalogT = 0;
+extern cameraReadInfo camread;
+
+static inline float
+vec_array(vec_float4 vec, unsigned int idx)
+{
+   union {
+      vec_float4 vec;
+      float      array[4];
+   } v;
+
+   v.vec = vec;
+
+   if (idx > 3)
+     return -1;
+   return v.array[idx];
+}
+
+int
+initMove()
+{
+   Spurs *spurs;
+   gemAttribute gem_attr;
+   u8 gem_spu_priorities[8] = { 1, 0, 0, 0, 0, 0, 0, 0 };
+   int ret;
+   int i;
+
+   spurs = initSpurs (NULL);
+
+   if (spurs == NULL)
+     goto error;
+
+   printf ("preparing GemAttribute structure with spurs\n");
+
+   initAttributeGem (&gem_attr, 1, NULL, spurs, gem_spu_priorities);
+   gem_attr->version = 2;
+   gem_attr->max = 1;
+   gem_attr->spurs = spurs;
+   gem_attr->memory = NULL;
+   gem_attr->spu_priorities[0] = 1;
+   for (i = 1; i < 8; ++i)
+     gem_attr->spu_priorities[i] = 0;
+
+   printf ("calling GemInit with GemAttribute structure version=%d max_connect=%d spurs=%X\n",
+           gem_attr.version, gem_attr.max, gem_attr.spurs);
+   ret = gemInit (&gem_attr);
+   printf ("return from GemInit %X \n", ret);
+   if (ret)
+     goto error;
+
+   ret = gemPrepareCamera (128, 0.5);
+   printf ("GemPrepareCamera return %d exposure set to 128 and quality to 0.5\n",
+           ret);
+   if (ret)
+     goto error;
+   ret = gemReset (0);
+   printf ("GemReset return %X \n", ret);
+   if (ret)
+     goto error;
+   return ret;
+
+error:
+   if (spurs)
+     endSpurs (spurs);
+   return NULL;
+}
+
+int
+endMove()
+{
+   endSpurs (spurs);
+   gemEnd ();
+   return 0;
+}
+
+int
+proccessGem(int t)
+{
+   int ret;
+
+   switch (t) {
+      case 0:
+
+        ret = gemUpdateStart (camread.buffer, camread.timestamp);
+
+        if (ret != 0)
+          {
+             printf ("Return from gemUpdateStart %X\n", ret);
+          }
+        break;
+
+      case 2:
+
+        ret = gemUpdateFinish ();
+        if (ret != 0)
+          {
+             printf ("Return from gemUpdateFinish %X\n", ret);
+          }
+        break;
+
+      case 3:
+        ret = gemConvertVideoFinish ();
+        if (ret != 0)
+          {
+             printf ("Return from gemConvertVideoFinish %X\n", ret);
+          }
+        break;
+
+      default:
+        ret = -1;
+        break;
+     }
+   return ret;
+}
+
+void
+readGemPad(int num_gem)
+{
+   gemState gem_state;
+   int ret;
+   unsigned int hues[] = { 4 << 24, 4 << 24, 4 << 24, 4 << 24 };
+   ret = gemGetState (0, 0, -22000, &gem_state);
+
+   newGemPad = gem_state.paddata.buttons & (~oldGemPad);
+   newGemAnalogT = gem_state.paddata.ANA_T;
+   oldGemPad = gem_state.paddata.buttons;
+
+   switch (ret) {
+      case 2:
+        gemForceRGB (num_gem, 0.5, 0.5, 0.5);
+        break;
+
+      case 5:
+
+        gemTrackHues (hues, NULL);
+        break;
+
+      default:
+        break;
+     }
+}
+
+void
+readGemAccPosition(int num_gem)
+{
+   vec_float4 position;
+
+   gemGetAccelerometerPositionInDevice (num_gem, &position);
+
+   printf (" accelerometer device coordinates [%f,%f,%f,%f]\n",
+           vec_array (position, 0), vec_array (position, 1), vec_array (position, 2),
+           vec_array (position, 3));
+}
+
+void
+readGemInertial(int num_gem)
+{
+   gemInertialState gem_inertial_state;
+   int ret;
+
+   ret = gemGetInertialState (num_gem, 0, -22000, &gem_inertial_state);
+   printf ("gemGetInertialState return %X\n", ret);
+   printf ("counter %d temperature %f\n", gem_inertial_state.counter,
+           gem_inertial_state.temperature);
+
+   printf (" accelerometer sensor   [%f,%f,%f,%f]\n",
+           vec_array (gem_inertial_state.accelerometer, 0),
+           vec_array (gem_inertial_state.accelerometer, 1),
+           vec_array (gem_inertial_state.accelerometer, 2),
+           vec_array (gem_inertial_state.accelerometer, 3));
+
+   printf (" accelerometer_bias sensor   [%f,%f,%f,%f]\n",
+           vec_array (gem_inertial_state.accelerometer_bias, 0),
+           vec_array (gem_inertial_state.accelerometer_bias, 1),
+           vec_array (gem_inertial_state.accelerometer_bias, 2),
+           vec_array (gem_inertial_state.accelerometer_bias, 3));
+
+   printf (" gyro sensor  [%f,%f,%f,%f]\n", vec_array (gem_inertial_state.gyro,
+                                                       0), vec_array (gem_inertial_state.gyro, 1),
+           vec_array (gem_inertial_state.gyro, 2),
+           vec_array (gem_inertial_state.gyro, 3));
+
+   printf (" gyro_bias sensor  [%f,%f,%f,%f]\n",
+           vec_array (gem_inertial_state.gyro_bias, 0),
+           vec_array (gem_inertial_state.gyro_bias, 1),
+           vec_array (gem_inertial_state.gyro_bias, 2),
+           vec_array (gem_inertial_state.gyro_bias, 3));
+}
+
+void
+readGem()
+{
+   gemState gem_state;
+   proccessGem (0);
+
+   proccessGem (2);
+
+   readGemPad (0); // This will read buttons from Move
+   switch (newGemPad) {
+      case 1:
+        printf ("Select pressed \n");
+        break;
+
+      case 2:
+        printf ("T pressed value %d\n", newGemAnalogT);
+        printf
+          ("Frame %d center of the sphere in world coordinates %f %f %f %f \n",
+          camread.frame, vec_array (gem_state.pos, 0), vec_array (gem_state.pos,
+                                                                  1), vec_array (gem_state.pos, 2), vec_array (gem_state.pos, 3));
+        break;
+
+      case 4:
+        printf ("Move pressed \n");
+        gemCalibrate (0);
+        break;
+
+      case 8:
+        printf ("Start pressed \n");
+        pos_x = 0;
+        pos_y = 0;
+        break;
+
+      case 16:
+        printf ("Triangle pressed \n");
+        getImageState ();
+        break;
+
+      case 32:
+        printf ("Circle pressed \n");
+        break;
+
+      case 64:
+        printf ("Cross pressed \n");
+        printf ("X,Y,Z position (mm) %f %f %f\n", vec_array (gem_state.pos, 0),
+                vec_array (gem_state.pos, 1), vec_array (gem_state.pos, 2));
+        readGemAccPosition (0);
+        break;
+
+      case 128:
+        printf ("Square pressed \n");
+        readGemInertial (0);
+        break;
+
+      default:
+        break;
+     }
+}
+
+void
+getImageState()
+{
+   int ret;
+
+   gemImageState imgState;
+
+   gemGetImageState (0, &imgState);
+   printf (" u  [%f]\n", imgState.u);
+   printf (" v  [%f]\n", imgState.v);
+   printf (" r  [%f]\n", imgState.r);
+   printf (" projectionx  [%f]\n", imgState.projectionx);
+   printf (" projectiony  [%f]\n", imgState.projectiony);
+   printf (" distance  [%f]\n", imgState.distance);
+   printf ("visible=%d r_valid=%d\n", imgState.visible, imgState.r_valid);
+   printf ("tiemestamp=%Ld\n", imgState.frame_time);
+}
+
diff -burN ecore-1.7.9/src/lib/ecore_psl1ght/gemutil.h ecore-1.7.9-PS3/src/lib/ecore_psl1ght/gemutil.h
--- ecore-1.7.9/src/lib/ecore_psl1ght/gemutil.h	1969-12-31 19:00:00.000000000 -0500
+++ ecore-1.7.9-PS3/src/lib/ecore_psl1ght/gemutil.h	2014-07-03 23:25:44.304875642 -0400
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) Antonio José Ramos Márquez (bigboss)
+ * Copyright (C) Youness Alaoui (KaKaRoTo)
+ */
+
+#ifndef __GEMUTIL_H__
+#define __GEMUTIL_H__
+
+#include <ppu-types.h>
+#include <spurs/spurs.h>
+#include <io/camera.h>
+#include <io/move.h>
+
+typedef struct
+{
+   Spurs *spurs;
+} moveContext;
+
+#endif /* __GEMUTIL_H__ */
diff -burN ecore-1.7.9/src/lib/ecore_psl1ght/moveutil.h ecore-1.7.9-PS3/src/lib/ecore_psl1ght/moveutil.h
--- ecore-1.7.9/src/lib/ecore_psl1ght/moveutil.h	1969-12-31 19:00:00.000000000 -0500
+++ ecore-1.7.9-PS3/src/lib/ecore_psl1ght/moveutil.h	2014-07-03 23:14:40.755852057 -0400
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) Antonio José Ramos Márquez (bigboss)
+ * Copyright (C) Youness Alaoui (KaKaRoTo)
+ */
+
+#ifndef __MOVEUTIL_H__
+#define __MOVEUTIL_H__
+
+#include <ppu-types.h>
+#include <spurs/spurs.h>
+#include <io/camera.h>
+#include <io/move.h>
+
+typedef struct
+{
+   Spurs         *spurs;
+   cameraInfoEx   camInfo;
+   cameraReadInfo camRead;
+   gemState       state;
+} moveContext;
+
+static inline float
+vec_array(vec_float4 vec, unsigned int idx)
+{
+   union {
+      vec_float4 vec;
+      float      array[4];
+   } v;
+
+   v.vec = vec;
+
+   if (idx > 3)
+     return -1;
+   return v.array[idx];
+}
+
+moveContext *initMove();
+void         endMove(moveContext *context);
+void         readGemState(moveContext *context);
+int          processMove(moveContext *context);
+void         moveGet3DPosition(moveContext *context, float *x, float *y, float *z);
+
+#endif /* __MOVEUTIL_H__ */
diff -burN ecore-1.7.9/src/lib/ecore_psl1ght/spursutil.h ecore-1.7.9-PS3/src/lib/ecore_psl1ght/spursutil.h
--- ecore-1.7.9/src/lib/ecore_psl1ght/spursutil.h	1969-12-31 19:00:00.000000000 -0500
+++ ecore-1.7.9-PS3/src/lib/ecore_psl1ght/spursutil.h	2014-07-03 23:14:40.755852057 -0400
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) Antonio José Ramos Márquez (bigboss)
+ * Copyright (C) Youness Alaoui (KaKaRoTo)
+ */
+
+#ifndef __SPURSUTIL_H__
+#define __SPURSUTIL_H__
+
+#include <ppu-types.h>
+#include <spurs/spurs.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Initialize spurs with a given name prefix (*/
+Spurs *initSpurs(const char *prefix_name);
+void   endSpurs(Spurs *spurs);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* __SPURSUTIL_H__ */
