diff -burN elementary-1.7.9/build.sh elementary-1.7.9-PS3/build.sh
--- elementary-1.7.9/build.sh	1969-12-31 19:00:00.000000000 -0500
+++ elementary-1.7.9-PS3/build.sh	2013-12-20 21:12:44.716322700 -0500
@@ -0,0 +1,11 @@
+#!/bin/sh
+## build.sh (By Spork Schivago)
+
+FSELF="make_self"
+
+cp src/bin/elementary_config elementary_config.elf
+sprxlinker elementary_config.elf
+$FSELF elementary_config.elf elementary_config.self
+cp src/bin/elementary_test elementary_test.elf
+sprxlinker elementary_test.elf
+$FSELF elementary_test.elf elementary_test.self
diff -burN elementary-1.7.9/configure.ac elementary-1.7.9-PS3/configure.ac
--- elementary-1.7.9/configure.ac	2013-10-07 17:13:03.000000000 -0400
+++ elementary-1.7.9-PS3/configure.ac	2013-12-20 20:12:56.890348815 -0500
@@ -469,9 +469,9 @@
 
 ELM_EDBUS_DEF="#undef"
 have_elementary_edbus="no"
-want_elementary_edbus="auto"
+want_elementary_edbus="no"
 AC_ARG_ENABLE([edbus],
-   [AC_HELP_STRING([--disable-edbus], [disable edbus support. @<:@default=detect@:>@])],
+   [AC_HELP_STRING([--enable-edbus], [enable edbus (v1) support. @<:@default=no@:>@])],
    [want_elementary_edbus=$enableval], [])
 
 if test "x$want_elementary_edbus" != "xno"; then
diff -burN elementary-1.7.9/Makefile.in elementary-1.7.9-PS3/Makefile.in
--- elementary-1.7.9/Makefile.in	2013-10-31 18:57:03.000000000 -0400
+++ elementary-1.7.9-PS3/Makefile.in	2013-12-20 20:57:42.486362900 -0500
@@ -95,7 +95,8 @@
 	$(srcdir)/Makefile.in $(srcdir)/Makefile.am \
 	$(top_srcdir)/configure $(am__configure_deps) \
 	$(srcdir)/elementary_config.h.in $(srcdir)/elementary.spec.in \
-	$(srcdir)/elementary.pc.in ABOUT-NLS COPYING TODO config.guess \
+	$(srcdir)/elementary.pc.in $(srcdir)/Elementary.h.in \
+	ABOUT-NLS COPYING TODO config.guess \
 	config.rpath config.sub depcomp install-sh missing ltmain.sh
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/ac_attribute.m4 \
diff -burN elementary-1.7.9/src/lib/Elementary.h elementary-1.7.9-PS3/src/lib/Elementary.h
--- elementary-1.7.9/src/lib/Elementary.h	2013-10-31 18:57:07.000000000 -0400
+++ elementary-1.7.9-PS3/src/lib/Elementary.h	2013-12-20 21:10:00.652163348 -0500
@@ -10,9 +10,9 @@
 #define ELM_UNIX
 #undef ELM_WIN32
 #undef ELM_WINCE
-#define ELM_EDBUS
-#define ELM_EFREET
-#define ELM_ETHUMB
+#undef ELM_EDBUS
+#undef ELM_EFREET
+#undef ELM_ETHUMB
 #undef ELM_WEB
 #undef ELM_EMAP
 #undef ELM_DEBUG
