diff -burN escape/src/lib/Escape.h escape-PS3/src/lib/Escape.h
--- escape/src/lib/Escape.h	2014-07-06 22:57:12.565027630 -0400
+++ escape-PS3/src/lib/Escape.h	2014-07-06 22:56:44.664026638 -0400
@@ -4,6 +4,11 @@
 #ifndef EAPI
 #define EAPI
 #endif /* EAPI */
+
+#ifdef __UNUSED__
+#undef __UNUSED__
+#endif
+
 #define __UNUSED__ __attribute__((unused))
 
 #include <unistd.h>
@@ -49,4 +54,7 @@
 }
 #endif
 
+/* To prevent implicit declaration of function */
+extern int closesocket(int socket);
+
 #endif /* __ESCAPE_H__ */
diff -burN escape/src/lib/escape_unistd.c escape-PS3/src/lib/escape_unistd.c
--- escape/src/lib/escape_unistd.c	2014-07-06 22:57:12.567027630 -0400
+++ escape-PS3/src/lib/escape_unistd.c	2014-07-06 22:20:36.422949572 -0400
@@ -104,7 +104,7 @@
    int socket2 = -1;
    fd_set read_set;
    fd_set write_set;
-   int len;
+   socklen_t len;
 
    temp = socket (AF_INET, SOCK_STREAM, 0);
 
Binary files escape/.svn/wc.db and escape-PS3/.svn/wc.db differ
